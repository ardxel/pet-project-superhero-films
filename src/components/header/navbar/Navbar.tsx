import React, { createRef, useEffect } from 'react';
import styles from './navbar.module.scss';
import { Link } from 'react-router-dom';
import { CrossSVG } from '@images/index';
import navLinks from '@constants/navLinks';

interface INavbarProps {
  isOpen: boolean;
  setIsOpen: (arg: boolean) => void;
}

export default function Navbar({ isOpen, setIsOpen }: INavbarProps) {
  return (
    <nav className={`${styles.nav} ${isOpen && styles.open}`}>
      <ul className={`${styles.links}`}>
        {navLinks.map((item, index) => {
          if (!isOpen && index === navLinks.length - 1) return;
          const { id, name, path } = item;
          return (
            <li key={id}>
              <Link to={path}>{name}</Link>
            </li>
          );
        })}
      </ul>
      {isOpen && (
        <button className={styles.closed} onClick={() => setIsOpen(false)}>
          <CrossSVG />
        </button>
      )}
    </nav>
  );
}
