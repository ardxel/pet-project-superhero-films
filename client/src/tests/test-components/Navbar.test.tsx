import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Navbar from '@components/header/navbar/Navbar';

describe('<Navbar> component test', () => {
  it('should render navbar for interaction with app', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Navbar
            isOpen
            setIsOpen={() => true}
            username={'john'}
          />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
