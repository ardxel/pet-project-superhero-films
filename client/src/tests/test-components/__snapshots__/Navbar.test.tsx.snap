// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar> component test should render navbar for interaction with app 1`] = `
<nav
  className="nav open"
>
  <ul
    className="links"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/news"
        onClick={[Function]}
      >
        News
      </a>
    </li>
    <li>
      <a
        href="/profile/john"
        onClick={[Function]}
      >
        Profile
      </a>
    </li>
  </ul>
  <button
    className="closed"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="CloseIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
  </button>
</nav>
`;
